app_dart/lib/src/request_handlers/push_engine_status_to_github.dart

```diff
@@ -50,18 +53,18 @@ class PushEngineStatusToGithub extends ApiRequestHandler<Body> {
     final LuciService luciService = luciServiceProvider(this);
     final Map<LuciBuilder, List<LuciTask>> luciTasks = await luciService.getRecentTasks(repo: 'engine');
 
-    String latestStatus = GithubBuildStatusUpdate.statusSuccess;
+    String status = GithubBuildStatusUpdate.statusSuccess;
     for (List<LuciTask> tasks in luciTasks.values) {
-      latestStatus = _getLatestStatus(tasks);
-      if (latestStatus == GithubBuildStatusUpdate.statusFailure) {
-        break;
+      final String latestStatus = await _getLatestStatus(tasks);
+      if (status == GithubBuildStatusUpdate.statusSuccess && latestStatus == GithubBuildStatusUpdate.statusFailure) {
+        status = GithubBuildStatusUpdate.statusFailure;
```

Comment:
Why are we overriding status?

Reply:
The status will only be overridden once: when latest build of any builder returns `failure`. Once overridden from `success` to `failure`, it will not be updated any more.